Windows CI build is failing

Windows CI builds are currently failing with:

```sh
[ 3%] Linking C executable blinky_m0.elf
 D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/arm-none-eabi/16.1.0/../../../../arm-none-eabi/bin/ld.exe: error: linker script file 'D:/a/hackrf/hackrf/firmware/libopencm3/lib\libopencm3_lpc43xx_m0.ld (D:/a/hackrf/hackrf/firmware/common/LPC43xx_M0_memory.ld)' appears multiple times
 collect2.exe: error: ld returned 1 exit status
```

The underlying cause is that `MSYS2` suddenly started shipping a version of `mingw-w64-ucrt-x86_64-arm-none-eabi-gcc` based on a development snapshot:

```sh
% arm-none-eabi-gcc --version
arm-none-eabi-gcc.exe (GCC) 16.1.0
```

Specifically, it's localized to Binutils (note the date):

```sh
% arm-none-eabi-ld --version
GNU ld (GNU Binutils) 2.47.20260726
Copyright (C) 2026 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
```

There's probably a regression in there somewhere, can also reproduce with:

```sh
% echo "MEMORY{}" > a.ld
% echo "SECTIONS{}" > b.ld
% arm-none-eabi-ld -T a.ld -T b.ld
C:\msys64\ucrt64\bin\arm-none-eabi-ld.exe: error: linker script file 'b.ld (a.ld)' appears multiple
times
```

Not really sure where to from here, maybe open an issue on MSYS2 referencing:

* https://github.com/msys2/MINGW-packages/pull/30638
* https://github.com/msys2/MINGW-packages/pull/30451 

In the meantime, can the MSYS2 GitHub action target a specific version of the `mingw-w64-ucrt-x86_64-arm-none-eabi-gcc` package for installation instead?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -284,6 +284,11 @@ jobs:
           mingw-w64-ucrt-x86_64-dfu-util
           mingw-w64-ucrt-x86_64-python-yaml
 
+    - name: Downgrade ARM linker (Windows, temporary)
+      if: matrix.os == 'windows'
+      shell: msys2 {0}
+      run: pacman -U --noconfirm https://repo.msys2.org/mingw/ucrt64/mingw-w64-ucrt-x86_64-arm-none-eabi-binutils-2.46.1-1-any.pkg.tar.zst
+
     - name: Build with MSYS (Windows)
       if: matrix.os == 'windows'
       shell: msys2 {0}

